packages/builder/src/types.zod.ts

```diff
@@ -0,0 +1,280 @@
+import { ethers } from 'ethers';
```

Comment:
This is really cool, having all the internal storage types in zod as well. Does this mean we will have a way to validate ipfs artifacts just to make sure? Can we use it as a way to verify that updates to cannon schema are non-breaking progmatically

Reply:
Yes! That's the idea, it's not fully implemented yet but the idea is that we can use zod as our source of truth which provides us inherent validation for our core types while being able to generate types directly from our zod schemas.